test/controllers/group_pair_controller_test.exs

```diff
@@ -0,0 +1,101 @@
+      %{conn: conn, logged_in_user: user, group: group} do
+      other_user = insert(:user)
+      pair = Pairmotron.TestHelper.create_pair([user, other_user], group)
+      create_retro(other_user, pair)
+      conn = get(conn, "/groups/#{group.id}/pairs")
+      assert html_response(conn, 200) =~ pair_retro_path(conn, :new, pair.id)
+    end
+
+    test "displays link to retro :show for pair and current user with retrospective",
+      %{conn: conn, logged_in_user: user, group: group} do
+      {_pair, retro} = create_pair_and_retro(user, group)
+      conn = get(conn, "/groups/#{group.id}/pairs")
+      assert html_response(conn, 200) =~ pair_retro_path(conn, :show, retro.id)
+    end
+
+    test "displays link to retro :show for pair and current user with retrospective for :show",
+      %{conn: conn, logged_in_user: user, group: group} do
+      {year, week} = Timex.iso_week(Timex.today)
+      pair = Pairmotron.TestHelper.create_pair([user], group, year, week)
+      retro = create_retro(user, pair) # create_retro function defines pair_date as Timex.today
+      conn = get conn, group_pair_path(conn, :show, group.id, year, week)
+      assert html_response(conn, 200) =~ pair_retro_path(conn, :show, retro.id)
+    end
+
+    test "cannot repairfy because user is not admin", %{conn: conn, group: group} do
+      {year, week} = Timex.iso_week(Timex.today)
+      conn = delete conn, group_pair_path(conn, :delete, group.id, year, week)
+      assert redirected_to(conn) == group_pair_path(conn, :show, group.id, year, week)
+    end
+  end
+
+  describe "while authenticated as admin" do
+    setup do
+      user = insert(:user, %{is_admin: true})
+      group = insert(:group, %{owner: user, users: [user]})
+      conn = build_conn() |> log_in(user)
+      {:ok, [conn: conn, logged_in_user: user, group: group]}
+    end
+
+    test "can repairify", %{conn: conn, group: group} do
+      {year, week} = Timex.iso_week(Timex.today)
+      conn = delete conn, group_pair_path(conn, :delete, group.id, year, week)
+      assert redirected_to(conn) == group_pair_path(conn, :show, group.id, year, week)
+    end
+
+  end
+
+  describe "while authenticated as a non-group user" do
+    setup do
+      user = insert(:user)
+      user2 = insert(:user)
+      group = insert(:group, %{owner: user2})
+      Pairmotron.TestHelper.create_pair([user2], group)
+      conn = build_conn() |> log_in(user)
+      {:ok, [conn: conn, logged_in_user: user, group: group]}
+    end
+
+    test "cannot access group pairs", %{conn: conn, group: group} do
+      conn = get conn, group_pair_path(conn, :show, group.id)
+      assert redirected_to(conn) == pair_path(conn, :index)
+    end
+
+    test "cannot access group pairs for a specific period", %{conn: conn, group: group} do
+      conn = get conn, group_pair_path(conn, :show, group.id, 2000, 1)
+      assert redirected_to(conn) == pair_path(conn, :show, 2000, 1)
+    end
```

Comment:
Good call. I'll add a test case to the `describe "while authenticated as admin" do` section.